scipy/spatial/tests/test_slerp.py

```diff
@@ -416,3 +403,64 @@ def test_interpolation_param_ndim(self, t):
             geometric_slerp(start=arr1,
                             end=arr1,
                             t=t)
+
+
+    @pytest.mark.parametrize("start, end, t, expected", [
+    # verify intuitive cases for t > 1 or < 0 (extrapolation)
+
+    # quarter way around the unit circle, doubling to
+    # half way around (allowed antipode since non-ambiguous
+    # geodesic):
+    ([0, 1], [1, 0], 2, [0, -1]),
+    # 1/8 way around the unit circle, quadrupling to half way
+    # (another allowed non-ambiguous antipode)
+    ([0, 1], [math.sqrt(2)/2, math.sqrt(2)/2], 4, [0, -1]),
+    # 1/8 way around the unit circle, 6x to 3/4 way around (270 deg)
+    ([0, 1], [math.sqrt(2)/2, math.sqrt(2)/2], 6, [-1, 0]),
+    # 1/8 way around the unit circle, 7x to 315 deg
+    ([0, 1], [math.sqrt(2)/2, math.sqrt(2)/2], 7, [-math.sqrt(2)/2, math.sqrt(2)/2]),
+    # the above three extrapolations in reverse order should be returned
+    # in that same order:
+    ([0, 1],
+     [math.sqrt(2)/2, math.sqrt(2)/2],
+     [7, 6, 4],
+     [[-math.sqrt(2)/2, math.sqrt(2)/2],
+      [-1, 0],
+      [0, -1]]),
+    # same case in 3d (sphere), in the xy plane:
+    ([0, 1, 0],
+     [math.sqrt(2)/2, math.sqrt(2)/2, 0],
+     [7, 6, 4],
+     [[-math.sqrt(2)/2, math.sqrt(2)/2, 0],
+      [-1, 0, 0],
+      [0, -1, 0]]),
+    # 1/8 way around the unit circle, moving 300 deg backwards
+    # should end up at pi/6 (since there is no forward travel with
+    # negative extrapolation)
+    ([0, 1], [math.sqrt(2)/2, math.sqrt(2)/2], (-300/45), [math.sqrt(3)/2, 0.5]),
+    ])
+    def test_extrapolation_basic(self, start, end, t, expected):
+        actual_path = geometric_slerp(start=start,
+                                      end=end,
+                                      t=t)
+        assert_allclose(actual_path, expected, atol=2e-16)
+
+    @pytest.mark.parametrize("start, end, t, expected", [
+    # cases where start and end proper are antipodes
+    # and t > 1 or < 0; these cases should issue a warning due to
+    # geodesic ambiguity
+
+    # North to South on unit sphere, then back to North again;
+    # there are an infinite number of possible routes, but they all
+    # return back to North
+    ([0, 0, 1], [0, 0, -1], -2, [0, 0, 1]),
+    # move "backwards" to South pole via an infinite number
+    # of possible routes
+    ([0, 0, 1], [0, 0, -1], -1, [0, 0, -1]),
+    ])
+    def test_extrapolation_antipodes(self, start, end, t, expected):
+        with pytest.warns(UserWarning, match='antipodes'):
```

Comment:
This doesn't look quite right. If extrapolation is supported, it should be working without warnings being emitted. If this warning filter is for a subset of the test cases, can you please split the list of tests into two?